templates/CMakeLists.txt.template

```diff
@@ -161,6 +162,9 @@
   include(cmake/gflags.cmake)
   include(cmake/benchmark.cmake)
 
+  include_directories(<%text>${CMAKE_CURRENT_SOURCE_DIR}/third_party/address_sorting</%text>)
```

Comment:
you shouldn't use `include_directories` as it adds the include blindly to all targets and that's a bad practice (and also it is redundant with adding to target_include_directories).

The clean way to do this is:
add cmake/address_sorting.cmake  (because you are adding a third party dependency, it deserves a separate file). The file should probably look similar to other cmake/*.cmake files  and should define `_gRPC_ADDRESS_SORTING_LIBRARIES`  (the library to depend on) and   `_gRPC_ADDRESS_SORTING_INCLUDE_DIR` (I did a cleanup recently so that all the third_party deps use the same style), which will be added to target_include_directories.  That way, the entire configuration and build of address_sorting is contained in a separate cmake file and the rest of the cmake builds just uses the variables defined there.

Reply:
Ok, because address_sorting is not a real submodule and it doesn't have it's own CMakeLists.txt, what I just sait might be incorrect. See my last comment for further explanation.

Reply:
Ok, two ways to do this:
1. you can either act that that address_sorting is not an external thing at all (from cmake's perspective) and try if the thing builds with only making changes to build.yaml and no manual changes to CMakeLists.txt
2. you can introduce cmake/address_sorting.cmake  which sets the _gRPC_ADDRESS_SORTING_* variables and explain the way address_sorting is being included there. CMakeLists.txt will only have minimal changes (using those variables in the right places) and no other plumbing.

Hope it helps.

Reply:
Thanks for comments, this is helpful :). I went with #2 here.

I believe that I can't rely solely on `build.yaml` only because there doesn't seem to be a way to add to the include path from the `build.yaml`, and `address_sorting.h` needs to be referenced with a `#include <address_sorting.h>` so that this can compile internally...